go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -182,6 +182,34 @@ type BaseNetworkController struct {
 	routeImportManager routeimport.Manager
 }
 
+func (oc *BaseNetworkController) reconcile(netInfo util.NetInfo, retryNodes []*kapi.Node) error {
+	reconcileRoutes := oc.routeImportManager != nil && oc.routeImportManager.NeedsReconciliation(netInfo)
+
+	err := util.ReconcileNetInfo(oc.ReconcilableNetInfo, netInfo)
+	if err != nil {
+		klog.Errorf("Failed to reconcile network %s: %v", oc.GetNetworkName(), err)
+	}
+
+	if reconcileRoutes {
+		err = oc.routeImportManager.ReconcileNetwork(oc.GetNetworkName())
+		if err != nil {
+			return err
```

Comment:
Technically we should just klog an error here, since `ReconcileNetInfo` above is a point of no return (I had a comment on the original code, perhaps you can add it back) as once it is called. `NeedsReconciliation` will return `false` anyway.